ai/src/test/kotlin/ktx/ai/BehaviorTreeTest.kt

```diff
@@ -0,0 +1,107 @@
+package ktx.ai
+
+import com.badlogic.gdx.ai.btree.BehaviorTree
+import com.badlogic.gdx.ai.btree.Task
+import io.kotlintest.mock.mock
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+
+class BehaviorTreeTest {
+
+  @Test
+  fun `behaviorTree function without arguments should return BehaviorTree object`() {
+    val behaviorTree = behaviorTree<Cat>()
+
+    assertNotNull(behaviorTree)
+  }
+
+  @Test
+  fun `behaviorTree function with non-null arguments should return BehaviorTree object with correct properties`() {
+    val rootTask = mock<Task<Cat>>()
+    val blackboard = mock<Cat>()
+    val tree = behaviorTree(rootTask, blackboard)
+
+    val rootTaskField = BehaviorTree::class.java.getDeclaredField("rootTask").apply {
```

Comment:
I'm assuming there's no way to check this without reflection? Could you perhaps test those based on some side effects of other functions? Relying on implementation details of a third party library should be avoided, but I understand if it's difficult to check otherwise.

Reply:
My bad, this mistake is very obvious. The reflection can actually be avoided by using public getters. Thanks for pointing this out.